File: HelloTuiLua/src/map/mapui.lua
require "extern"
require "tagMap/Tag_map"

Mapui = class("Mapui",function()
	return TuiBase:create()
end)

Mapui.__index = Mapui
local __instance = nil

function Mapui:create()
	local ret = Mapui.new()
	__instance = ret
	ret:setOnLoadSceneScriptHandler(function() ret:onLoadScene() end)
	return ret
end

function Mapui:getControl(tagPanel,tagControl)
	local ret = nil
	ret = self:getPanel(tagPanel):getChildByTag(tagControl)
	return ret
end

function Mapui:getPanel(tagPanel)
	local ret = nil
	if tagPanel == Tag_map.PANEL_MAP then
		ret = self:getChildByTag(tagPanel)
	end
	return ret
end

local function event_btn_back(p_sender)
	
	CSceneManager:getInstance():replaceScene(
		CCSceneExTransitionFade:create(0.5,LoadScene("Welcomeui")))
end


function Mapui:onLoadScene()

	TuiManager:getInstance():parseScene(self,"panel_map",PATH_MAP)

	local mapView = self:getControl(Tag_map.PANEL_MAP,Tag_map.MAP_TEST)
	mapView:setRole(cc.Sprite:create("map/img_icon.png"),cc.p(0,0),5)
    
    local btnBack = self:getControl(Tag_map.PANEL_MAP,Tag_map.BTN_BACK)
	btnBack:setOnClickScriptHandler(event_btn_back)
end

File: HelloTuiCpp/Classes/scene/Map/Mapui.cpp
#include "Mapui.h"
#include "tui/TuiManager.h"
#include "tui/tagMap/Tag_map.h"

void Mapui::onLoadScene()
{
	TuiManager::getInstance()->parseScene(this,"panel_map",PATH_MAP);

    CMapView* pMapView = (CMapView*)this->getControl(PANEL_MAP,MAP_TEST);
	pMapView->setRole(Sprite::create("map/img_icon.png"),Vec2(0,0),1);
    
    CButton* pBtnBack = (CButton*)this->getControl(PANEL_MAP,BTN_BACK);
    pBtnBack->setOnClickListener(this, ccw_click_selector(Mapui::event_btn_back));
}

void Mapui::event_btn_back(Ref* pSender)
{
    CSceneManager::getInstance()->replaceScene(CCSceneExTransitionFade::create(0.5f, LoadScene("Main::Mainui")));
}


Node* Mapui::getPanel( int tagPanel )
{
    Node *pPanel = nullptr;
    switch (tagPanel)
    {
        case PANEL_MAP:
            pPanel = this->getChildByTag(PANEL_MAP);
            break;
    }
    return pPanel;
}